cmd/storj/cmd/cp.go

```diff
@@ -32,79 +37,164 @@ func init() {
 	cpCmd.Flags().String("config", filepath.Join(defaultConfDir, "config.yaml"), "path to configuration")
 }
 
-func copy(cmd *cobra.Command, args []string) (err error) {
-	ctx := process.Ctx(cmd)
+// upload uploads args[0] from local machine to s3 compatible object args[1]
+func upload(ctx context.Context, bs buckets.Store, srcFile string, destObj *url.URL) error {
+	if destObj.Scheme == "" {
+		fmt.Println("Invalid destination")
+		return nil
+	}
 
-	if len(args) == 0 {
-		return errs.New("No file specified for copy")
+	// if object name not specified, default to filename
+	if destObj.Path == "" || destObj.Path == "/" {
```

Comment:
this should be the case if the path ends in a slash at all, right? add the filename after the slash?

Reply:
Done